variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/factories/VariantVcfFactory.java

```diff
@@ -351,4 +361,38 @@ protected static int mapToMultiallelicIndex(int parsedAllele, int numAllele) {
         }
         return correctedAllele;
     }
+
+    private void checkVariantInformation(Variant variant, String fileId,
```

Comment:
I would split this method in 2:

* `isNonVariant`
* `canAlleleFrequenciesBeCalculated`

It should be possible to reuse part of the existing code, while reducing the interleaving of the 2 checks. For instance, right now `checkAttributeIsNotZero` mixes both checking whether the attribute is present, then whether it is zero. One exception is thrown from inside the method, and the other has to be checked from the calling method.